File: examples/python/end_user/list_end_users.py
# Usage: uv run python end_user/list_end_users.py

import asyncio
from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        try:
            # List all end users with default pagination (20 per page)
            result = await cdp.end_user.list_end_users(
                page_size=2,
                sort=["createdAt=asc"]
            )
            
            print(f"Found {len(result.end_users)} end users on first page")
            for end_user in result.end_users:
                print(f"  - User ID: {end_user.user_id}")
                print(f"    EVM Accounts: {end_user.evm_accounts}")
                print(f"    EVM Smart Accounts: {end_user.evm_smart_accounts}")
                print(f"    Solana Accounts: {end_user.solana_accounts}")
                print()

        except Exception as e:
            print(f"Error listing end users: {e}")
            raise e


asyncio.run(main())

File: examples/java/src/main/java/com/coinbase/cdp/examples/enduser/ListEndUsers.java
package com.coinbase.cdp.examples.enduser;

import com.coinbase.cdp.CdpClient;
import com.coinbase.cdp.client.enduser.EndUserClientOptions.ListEndUsersOptions;
import com.coinbase.cdp.examples.utils.EnvLoader;
import java.util.List;

/**
 * Example: List end users with pagination.
 *
 * <p>This example demonstrates how to list end users sorted by creation date in descending order,
 * with a page size of 10.
 *
 * <p>Usage: ./gradlew runListEndUsers
 */
public class ListEndUsers {

  public static void main(String[] args) throws Exception {
    EnvLoader.load();

    try (CdpClient cdp = CdpClient.create()) {
      System.out.println("Listing end users (page size: 10, sorted by createdAt desc)...");
      System.out.println();

      var response =
          cdp.endUser()
              .listEndUsers(
                  ListEndUsersOptions.builder()
                      .pageSize(10)
                      .sort(List.of("createdAt=desc"))
                      .build());

      var endUsers = response.getEndUsers();
      System.out.println("End Users (" + endUsers.size() + " returned):");
      System.out.println();

      for (var endUser : endUsers) {
        System.out.println("  User ID: " + endUser.getUserId());
        System.out.println("    EVM accounts: " + endUser.getEvmAccountObjects().size());
        System.out.println(
            "    EVM smart accounts: " + endUser.getEvmSmartAccountObjects().size());
        System.out.println(
            "    Solana accounts: " + endUser.getSolanaAccountObjects().size());
        System.out.println();
      }

      if (endUsers.isEmpty()) {
        System.out.println("  No end users found. Run CreateEndUser first.");
      }
    }
  }
}
